[EGD-8148] Download asset documentation update (#3462)
mudita/MuditaOS: doc/build_targets.md

|common| \<Product\>                    |                                                   | Binary target for the product |
|common| \<Product\>-disk-img           | \<Product\>.img                                   | Disk image for the product    |
|RT1051| \<Product\>-StandaloneImage    | PurePhone-\<version\>-RT1051-package-standalone   | Creates image that can be `dd` or `pureflash` to the device|
|RT1051| \<Product>\-UpdatePackage      | PurePhone-\<version\>-RT1051-Update.tar           | Creates Update package, that can be used by Mudita Center or an update script|
|linux | check                          |                                                   | build and run unittests |
|common| assets                         |                                                   | Mudita company private assets |
|common| json-images-target             |                                                   | Mudita company public assets |
|RT1051| ecoboot.bin                    |                                                   | OS bootloader |
|RT1051| updater.bin                    |                                                   | OS updater |

## Product binary target

> Just software binary be loaded either by:

Reviewer: ```suggestion
The software binary can be loaded by:
```

> Just software binary be loaded either by:

Author: outhdated :)

- JLink
- Directly onto the device by:
    - Updater utility
    - direct memory access

> Devices in field have keys burned, therefore additional step - signing them will be required. Without signature devices will show warning on boot.

Reviewer: ```suggestion
Devices in the field have keys burned, therefore an additional step - binary signing will be required. Without the signature, devices will show a warning on boot.
```

## Disk image

> Image of software to be loaded onto the device memory. Has all neccessary data to properly run firmware. 

Reviewer: ```suggestion
Image of the software to be loaded onto the device memory. Has all necessary data to properly run the OS. 
```

> Disk image can be build with addtional, development assets. Please look for `WITH_DEVELOPMENT_FEATURES` for more information in make.

Reviewer: ```suggestion
The disk image can be built with additional, development assets. Please look for `WITH_DEVELOPMENT_FEATURES` for more information in make.
```

## Update package

> Tar-file meant to update software with the updater utility. It has all necessary update assets and metadata.

Reviewer: ```suggestion
Tar-file meant to update software with the updater utility. It has all the necessary update assets and metadata.
```

> Currently update package has all base data that should be on the device, though to make updates smaller minimal updates packages, as well gzipped packages should be easily possible.

Reviewer: ```suggestion
Currently, the update package has all of the base data that should be on the device, though to make updates smaller, minimal updates packages, as well as gzipped packages should be easily possible.
```

## Check

> Builds and executes all unittests in the repository for selected product.

Reviewer: ```suggestion
Builds and executes all of the unit tests in the repository for the selected product.
```

> Uses [download_assets](./download_assets.md) unittests required assets (fonts)

Reviewer: ```suggestion
Uses [download_assets](./download_assets.md) to download unit test's required assets (fonts)
```

## assets

Downloads Mudita private assets with [download_assets](./download_assets.md) that company is either not eligible or capable of sharing publicly.
These are distributed with the image and update targets available for each product and. includes i.e. fonts.

## json-images-target

Downloads Mudita public assets with [download_assets](./download_assets.md).

## ecoboot.bin

Downloads the bootloader with [download_assets](./download_assets.md).

> OS bootloader, used to pre-init hardware and launch either OS or updater utility, uses

Reviewer: uses what? :P

## updater.bin

Downloads the updater with [download_assets](./download_assets.md).

> Updater is used to perform firmware update via update packages. 

Reviewer: ```suggestion
Updater is used to perform the firmware upgrade via update packages. 
```
